test/optimizations/constDomain/commaDeclLeak.chpl

```diff
@@ -0,0 +1,20 @@
+// For the cases below, callDestructors removes initCopys while avoiding the
+// destruction for those domains. However, constant domain optimization's AST
+// modifications interfered with the AST pattern that resulted in some initCopys
+// to be not dropped. This caused even numbered domains below to leak:
```

Comment:
Since this is a test checking for a leak, should it have a compopts with `--memLeaks`?

Reply:
Makes sense. Changed.